app/controllers/notifications_controller.rb

```diff
@@ -1,11 +1,19 @@
 class NotificationsController < ApplicationController
   def index
-    page = (params[:page] || 1).to_i
-    @notifications = current_user.notifications.page(page).per(20)
-  end
-
-  def unseen
-    render json: current_user.notifications.unseen
+    respond_to do |format|
+      format.html do
+        page = (params[:page] || 1).to_i
+        @notifications = current_user.notifications.page(page).per(20)
+      end
+      format.json do
+        notifications = current_user.notifications.unseen
```

Comment:
if we call `.to_a` on the unseen ones, and then just `notifications = notifications + seen` should join the two arrays and you don’t need to iterate